src/pages/chat/ChatRoomPage.tsx

```diff
@@ -1,35 +1,51 @@
+import withProviders from "react-with-providers";
+
 import { Screen } from "@/apps/Screen";
 
 import { ChatHistoryGroup } from "@/entities/chat/ui/ChatHistory/ChatHistoryGroup";
 import { ChatHistoryItem } from "@/entities/chat/ui/ChatHistory/ChatHistoryItem";
 import { ChatHistoryTime } from "@/entities/chat/ui/ChatHistory/ChatHistoryTime";
 import { ChatInput } from "@/entities/chat/ui/ChatInput/ChatInput";
 import { ChatNavTop } from "@/entities/chat/ui/ChatNavTop/ChatNavTop";
+import { ChatHistoryContextProvider } from "@/features/chat/contexts/ChatHistoryContext";
+import { useChat } from "@/features/chat/hooks/useChat";
 import { ChatGradientLayer } from "@/shared/components/GradientLayers/ChatGradientLayer";
 
-export default function ChatRoomPage() {
+export default withProviders([<ChatHistoryContextProvider />], function ChatRoomPage() {
+    const { chatInputRef, sendMessage, chatHistory } = useChat({
+        userId: 10,
+        roomId: 1,
+    });
+
     return (
         <Screen>
-            <ChatGradientLayer className="w-full min-h-screen">
-                <ChatNavTop title={"채팅방 이름"} currentQuota={2} maxQuota={4} />
+            <ChatHistoryContextProvider>
+                <ChatGradientLayer className="w-full min-h-screen">
+                    <ChatNavTop title={"채팅방 이름"} currentQuota={2} maxQuota={4} />
```

Comment:
이 부분은 나중에 ChatRoom 인원보고 할꺼죠?
```suggestion
maxQuota={MAX_PARTICIPANTS}
```
로 매직넘버 없애도 될거같아요